inference-engine/thirdparty/clDNN/src/include/loop_inst.h

```diff
@@ -99,19 +86,95 @@ struct typed_program_node<loop> : public typed_program_node_base<loop> {
 
     static size_t convert_to_raw_axis(size_t axis, size_t ndim) {
         // convert between bfyx, bfzyx, bfzyxw and tensor.size.raw
-        assert(axis < ndim);
+        if (axis >= ndim) {
+            throw std::runtime_error("axis should be less than ndim");
+        }
+
         if (axis < 2) {
             return axis;
         }
         return (ndim - 1) - (axis - 2);
     }
 
+    // read scala value from data primitive
+    static int64_t read_scalar_value(memory::ptr mem, stream& stream) {
+        int64_t trip_count = 0;
+        const layout& prim_layout = mem->get_layout();
+
+        switch (prim_layout.data_type) {
+        case data_types::u8: {
+            mem_lock<uint8_t> lock_prim_output{mem, stream};
+            trip_count = *lock_prim_output.data();
+            break;
+        }
+        case data_types::i8: {
+            mem_lock<int8_t> lock_prim_output{mem, stream};
+            trip_count = *lock_prim_output.data();
+            break;
+        }
+        case data_types::i32: {
+            mem_lock<int32_t> lock_prim_output{mem, stream};
+            trip_count = *lock_prim_output.data();
+            break;
+        }
+        case data_types::i64: {
+            mem_lock<int64_t> lock_prim_output{mem, stream};
+            trip_count = *lock_prim_output.data();
+            break;
+        }
+        default:
+            throw std::runtime_error("Invalid data type : " + data_type_traits::name(prim_layout.data_type));
+        }
+        return trip_count;
+    }
+
+    template<typename T>
+    static inline void validate_input_value(int64_t input) {
+        if (input < std::numeric_limits<T>::min() || input > std::numeric_limits<T>::max()) {
+            throw std::runtime_error("Invalid data value : " + std::to_string(input));
+        }
+    }
+
+    static void write_scalar_value(memory::ptr mem, stream& stream, int64_t input) {
+        const layout& prim_layout = mem->get_layout();
+
+        switch (prim_layout.data_type) {
+        case data_types::u8: {
+            validate_input_value<uint8_t>(input);
+            mem_lock<uint8_t> lock_prim_output{mem, stream};
+            *lock_prim_output.data() = static_cast<uint8_t>(input);
```

Comment:
you can use `operator[]` and have shorter form: `lock_prim_output[0] = ...`

Reply:
Fixed it. thanks.